pennylane/pulse/transmon.py

```diff
@@ -215,3 +217,263 @@ def __add__(self, other):
+
+    **Example**
+
+    We can construct a drive term acting on qubit ``0`` in the following way. We parametrize the amplitude and phase
+    via :math:`\Omega(t) = \Omega \sin(\pi t)` and :math:`\phi(t) = \phi (t - \frac{1}{2})`.
+
+    .. code-block:: python3
+
+        def amp(Omega, t): return Omega * jnp.sin(jnp.pi*t)
+        def phase(phi, t): return phi * (t - 0.5)
+
+        H = qml.pulse.transmon_drive(amp, phase, 0, 0)
+
+        t = 0.5
+        Omega = 0.1
+        phi = 0.001
+        params = [Omega, phi]
+
+    Evaluated at :math:`t = \frac{1}{2}` with the parameters :math:`\Omega = 0.1` and :math:`\phi = 10^{-3}` we obtain
+    :math:`\Omega \left(\frac{1}{2}(\sigma^x + i \sigma^y) + \frac{1}{2}(\sigma^x + i \sigma^y)\right) = \Omega \sigma^x`.
+
+    >>> H(params, t)
+    (0.1*(PauliX(wires=[0]))) + (0.0*(-1*(PauliY(wires=[0]))))
+
+    We can combine ``transmon_drive()`` with :func:`~.transmon_interaction` to create a full driven transmon Hamiltonian.
+    Let us look at a chain of three transmon qubits that are coupled with their direct neighbors. We provide all numbers in
+    units of :math:`2\pi\text{GHz}`. We parametrize the amplitude as a sinusodial and make the maximum amplitude
+    as well as the drive frequency trainable parameters. We simulate the evolution for a time window of :math:`[0, 5]\text{ns}`.
+
+    .. code-block:: python3
+
+        omega = [5.1, 5., 5.3]
+        connections = [[0, 1], [1, 2]]
+        g = [0.02, 0.05]
+        H = qml.pulse.transmon_interaction(omega, connections, g, wires=range(3))
+
+        def amp(Omega, t): return Omega * jnp.sin(t)
+        def freq(fr, t): return fr
+        phase = 0.
+        t=2
+
+        H += qml.pulse.transmon_drive(amp, phase, freq, 0)
+        H += qml.pulse.transmon_drive(amp, phase, freq, 1)
+        H += qml.pulse.transmon_drive(amp, phase, freq, 2)
+
+        dev = qml.device("default.qubit.jax", wires=range(3))
+
+        @jax.jit
+        @qml.qnode(dev, interface="jax")
+        def qnode(params):
+            qml.evolve(H)(params, t=5.)
+            return qml.expval(qml.PauliZ(0) + qml.PauliZ(1) + qml.PauliZ(2))
+
+    We evaluate the Hamiltonian with some arbitrarily chosen maximum amplitudes and set
+    the drive frequency equal to the qubit frequencies. Note how the order of the construction
+    of ``H`` determines the order with which the parameters need to be passed to
+    :class:`~.ParametrizedHamiltonian` and :func:`~.evolve`. We made the drive frequencies
+    trainable parameters by providing a constant callable above instead of fixed values.
+    This allows us to differentiate with respect to the frequencies and to optimize them.
+
+    >>> Omega0, Omega1, Omega2 = [0.5, 0.3, 0.6]
+    >>> fr0, fr1, fr2 = omega
+    >>> params = [Omega0, fr0, Omega1, fr1, Omega2, fr2]
+    >>> qnode(params)
+    Array(2.25098131, dtype=float64)
```

Comment:
Missing blank line after this output, so it renders as a single large block with outputs in the middle

Reply:
What do you mean? 
![image](https://user-images.githubusercontent.com/43949391/233120593-a006354f-233f-4999-840d-a7e58ba40e76.png)

Reply:
@lillian542 I believe this is formatted correctly, unless I am missing a subtlety :) The way the copy button works for `pycon` codeblocks is that it only copies _input_ lines, and removes the `>>>`.

This is deliberate, so that you can paste it in a python console and will see the same output

Reply:
E.g., copying the codeblock gives
```
Omega0, Omega1, Omega2 = [0.5, 0.3, 0.6]
fr0, fr1, fr2 = omega
params = [Omega0, fr0, Omega1, fr1, Omega2, fr2]
qnode(params)
jax.grad(qnode)(params)
```

Reply:
Ah, okay! I was under the impression that you only want to copy up to an output, so that each block in the examples should be a notebook cell. If you want the output here, you have to split this block into two cells in Jupyter. As it is, when you copy, you get the code block you show above, which effectively skips the line `qnode(params)` if you just copy-paste-run, and you'll only see the second output.

Reply:
Got it. I think the overwhelming standard in the docs is to have a single pycon codeblock with multiple interspersed outputs, so recommend sticking to that just for consistency!

Reply:
@lillian542 yep Jupyter is an outlier with pycon syntax. Most other environments that take pycon syntax (the python console, ipython, etc.) will show all intermediate outputs when you press enter.

It has always bothered me that Jupyter was non-standard this way 😢

Reply:
Interesting! Its the only thing I've ever used, so I had no idea... sounds good!